rs/state_manager/src/split.rs

```diff
@@ -170,7 +171,7 @@ fn read_checkpoint(
 /// Writes the given `ReplicatedState` into a new checkpoint under
 /// `state_layout`, based off of `old_cp`.
 fn write_checkpoint(
-    state: &ReplicatedState,
+    mut state: ReplicatedState,
```

Comment:
Wouldn't a `&mut ReplicatedState` be sufficient here? Just so no cloning would be needed in case someone may want to do something more with the `ReplicatedState` afterwards.

Reply:
You're right, fixed.